app/consumer.py

```diff
@@ -56,22 +56,62 @@ def on_message(self, unused_channel, basic_deliver, properties, body):
             tx_id=tx_id,
         )
 
-        document = get_doc_from_store(tx_id)
+        try:
+            document = get_doc_from_store(tx_id)
+            self.process(basic_deliver, delivery_count, document)
+
+        except NotFoundError as e:
+            logger.error("Document not found",
+                         exception=e,
+                         delivery_count=delivery_count,
+                         tx_id=tx_id,
+                         )
+
+        except BadRequestError as e:
+            self.reject_message(basic_deliver.delivery_tag, tx_id=tx_id)
+            logger.error("Bad Request",
```

Comment:
Should probably be logging what you're just _about_ to do rather than what you're _done_. That way if the thing you're doing dies and fails to return the log will still be written for traceability.

So e.g
```
logger.error("Bad request", action="rejecting" ... )
self.reject_message( ... )
```